File: python/price_tracking.py
"""
CSMarketAPI Price Tracking Example

Build a price tracker that monitors specific CS2 items and detects price changes.
"""
import asyncio
import os
from datetime import datetime
from csmarketapi import CSMarketAPI
from csmarketapi.enums import Market, Currency

# Items to track
WATCHLIST = [
    "Chroma 2 Case",
    "AWP | Asiimov (Field-Tested)",
    "AK-47 | Redline (Field-Tested)",
    "Glove Case",
]

# Price alerts: item -> target price
PRICE_ALERTS = {
    "Chroma 2 Case": 0.40,
    "AWP | Asiimov (Field-Tested)": 35.00,
    "AK-47 | Redline (Field-Tested)": 15.00,
}

CHECK_INTERVAL = 300  # 5 minutes


async def get_item_prices(client, item_name):
    """Fetch current prices for an item across all markets."""
    res = await client.get_listings_latest_aggregated(
        market_hash_name=item_name,
        markets=list(Market),
        currency=Currency.USD
    )
    return res


async def track_watchlist(client):
    """Track prices for all items in watchlist."""
    print("=" * 60)
    print("CS2 PRICE TRACKER")
    print("=" * 60)

    for item_name in WATCHLIST:
        try:
            res = await client.get_listings_latest_aggregated(
                market_hash_name=item_name,
                markets=list(Market),
                currency=Currency.USD
            )

            if res.listings:
                # Find lowest price
                lowest = min(res.listings, key=lambda x: x.min_price)
                avg_price = sum(l.min_price for l in res.listings) / len(res.listings)

                print(f"\n{item_name}")
                print(f"  Lowest: ${lowest.min_price:.2f} on {lowest.market}")
                print(f"  Average: ${avg_price:.2f} across {len(res.listings)} markets")
            else:
                print(f"\n{item_name}: No listings found")

        except Exception as e:
            print(f"\n{item_name}: Error - {e}")


async def check_alerts(client):
    """Check for price drops and trigger alerts."""
    alerts_triggered = []

    for item_name, target_price in PRICE_ALERTS.items():
        try:
            res = await client.get_listings_latest_aggregated(
                market_hash_name=item_name,
                markets=list(Market),
                currency=Currency.USD
            )

            if res.listings:
                lowest = min(res.listings, key=lambda x: x.min_price)

                if lowest.min_price <= target_price:
                    alerts_triggered.append({
                        "item": item_name,
                        "price": lowest.min_price,
                        "target": target_price,
                        "market": lowest.market,
                    })

        except Exception as e:
            print(f"Error checking {item_name}: {e}")

    return alerts_triggered


async def run_periodic_check(client):
    """Run price checks on a schedule."""
    print("Starting price tracker. Press Ctrl+C to stop.")

    while True:
        timestamp = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
        print(f"\n[{timestamp}] Checking prices...")

        for item_name in WATCHLIST[:2]:  # Check first 2 items for demo
            try:
                res = await client.get_listings_latest_aggregated(
                    market_hash_name=item_name,
                    markets=list(Market),
                    currency=Currency.USD
                )
                if res.listings:
                    lowest = min(res.listings, key=lambda x: x.min_price)
                    print(f"  {item_name}: ${lowest.min_price:.2f} ({lowest.market})")
            except Exception as e:
                print(f"  {item_name}: Error - {e}")

        await asyncio.sleep(CHECK_INTERVAL)


async def main():
    api_key = os.environ["CSMARKETAPI_KEY"]

    async with CSMarketAPI(api_key) as client:
        # Run watchlist tracker
        await track_watchlist(client)

        # Check price alerts
        print("\n" + "=" * 60)
        print("CHECKING PRICE ALERTS")
        print("=" * 60)

        alerts = await check_alerts(client)

        if alerts:
            print("\nPrice alerts triggered!")
            for alert in alerts:
                savings = alert["target"] - alert["price"]
                print(f"  {alert['item']}")
                print(f"    Price: ${alert['price']:.2f} (target: ${alert['target']:.2f})")
                print(f"    Market: {alert['market']}")
                print(f"    Savings: ${savings:.2f}")
        else:
            print("\nNo alerts triggered. All prices above targets.")


if __name__ == "__main__":
    try:
        asyncio.run(main())
    except KeyboardInterrupt:
        print("\nStopped.")

File: javascript/price_tracking.js
/**
 * CSMarketAPI Price Tracking Example
 *
 * Build a price tracker that monitors specific CS2 items and detects price changes.
 */
import { CSMarketAPI, Market, Currency } from "csmarketapi"

// Items to track - replace with your items
const WATCHLIST = [
  "Chroma 2 Case",
  "AWP | Asiimov (Field-Tested)",
  "AK-47 | Redline (Field-Tested)",
  "Glove Case",
]

// Price alerts: item -> target price - customize your alerts
const PRICE_ALERTS = {
  "Chroma 2 Case": 0.40,
  "AWP | Asiimov (Field-Tested)": 35.00,
  "AK-47 | Redline (Field-Tested)": 15.00,
}

const CHECK_INTERVAL = 300_000 // 5 minutes in ms

async function getItemPrices(client, itemName) {
  return await client.getListingsLatestAggregated({
    marketHashName: itemName,
    markets: Object.values(Market),
    currency: Currency.USD,
  })
}

async function trackWatchlist(client) {
  console.log("=".repeat(60))
  console.log("CS2 PRICE TRACKER")
  console.log("=".repeat(60))

  for (const itemName of WATCHLIST) {
    try {
      const res = await client.getListingsLatestAggregated({
        marketHashName: itemName,
        currency: Currency.USD,
      })

      if (res.listings.length > 0) {
        const lowest = res.listings.reduce((a, b) =>
          a.min_price < b.min_price ? a : b
        )
        const avgPrice = res.listings.reduce((sum, l) =>
          sum + l.min_price, 0
        ) / res.listings.length

        console.log(`\n${itemName}`)
        console.log(`  Lowest: $${lowest.min_price.toFixed(2)} on ${lowest.market}`)
        console.log(`  Average: $${avgPrice.toFixed(2)} across ${res.listings.length} markets`)
      } else {
        console.log(`\n${itemName}: No listings found`)
      }
    } catch (error) {
      console.log(`\n${itemName}: Error - ${error.message}`)
    }
  }
}

async function checkAlerts(client) {
  const alertsTriggered = []

  for (const [itemName, targetPrice] of Object.entries(PRICE_ALERTS)) {
    try {
      const res = await client.getListingsLatestAggregated({
        marketHashName: itemName,
        currency: Currency.USD,
      })

      if (res.listings.length > 0) {
        const lowest = res.listings.reduce((a, b) =>
          a.min_price < b.min_price ? a : b
        )

        if (lowest.min_price <= targetPrice) {
          alertsTriggered.push({
            item: itemName,
            price: lowest.min_price,
            target: targetPrice,
            market: lowest.market,
          })
        }
      }
    } catch (error) {
      console.log(`Error checking ${itemName}: ${error.message}`)
    }
  }

  return alertsTriggered
}

// Main execution
// Get API key from environment variable
// Set it with: export CSMARKETAPI_KEY="your_api_key"
const client = new CSMarketAPI({
  apiKey: process.env.CSMARKETAPI_KEY
})

// Run watchlist tracker
await trackWatchlist(client)

// Check price alerts
console.log("\n" + "=".repeat(60))
console.log("CHECKING PRICE ALERTS")
console.log("=".repeat(60))

const alerts = await checkAlerts(client)

if (alerts.length > 0) {
  console.log("\nPrice alerts triggered!")
  for (const alert of alerts) {
    const savings = alert.target - alert.price
    console.log(`  ${alert.item}`)
    console.log(`    Price: $${alert.price.toFixed(2)} (target: $${alert.target.toFixed(2)})`)
    console.log(`    Market: ${alert.market}`)
    console.log(`    Savings: $${savings.toFixed(2)}`)
  }
} else {
  console.log("\nNo alerts triggered. All prices above targets.")
}
